src/Tools/dotnet-dump/Dumper.Windows.cs

```diff
@@ -15,61 +15,57 @@ public partial class Dumper
-                return Task.Run(() =>
+                // Open the file for writing
+                using (var stream = new FileStream(outputFile, FileMode.Create, FileAccess.ReadWrite, FileShare.None))
                 {
-                    // Open the file for writing
-                    using (var stream = new FileStream(outputFile, FileMode.Create, FileAccess.ReadWrite, FileShare.None))
+                    var exceptionInfo = new NativeMethods.MINIDUMP_EXCEPTION_INFORMATION();
+
+                    NativeMethods.MINIDUMP_TYPE dumpType = NativeMethods.MINIDUMP_TYPE.MiniDumpNormal;
+                    switch (type)
                     {
-                        var exceptionInfo = new NativeMethods.MINIDUMP_EXCEPTION_INFORMATION();
+                        case DumpTypeOption.Full:
+                            dumpType = NativeMethods.MINIDUMP_TYPE.MiniDumpWithFullMemory |
+                                       NativeMethods.MINIDUMP_TYPE.MiniDumpWithDataSegs |
+                                       NativeMethods.MINIDUMP_TYPE.MiniDumpWithHandleData |
+                                       NativeMethods.MINIDUMP_TYPE.MiniDumpWithUnloadedModules |
+                                       NativeMethods.MINIDUMP_TYPE.MiniDumpWithFullMemoryInfo |
+                                       NativeMethods.MINIDUMP_TYPE.MiniDumpWithThreadInfo |
+                                       NativeMethods.MINIDUMP_TYPE.MiniDumpWithTokenInformation;
+                            break;
+                        case DumpTypeOption.Heap:
+                            dumpType = NativeMethods.MINIDUMP_TYPE.MiniDumpWithPrivateReadWriteMemory |
+                                       NativeMethods.MINIDUMP_TYPE.MiniDumpWithDataSegs |
+                                       NativeMethods.MINIDUMP_TYPE.MiniDumpWithHandleData |
+                                       NativeMethods.MINIDUMP_TYPE.MiniDumpWithUnloadedModules |
+                                       NativeMethods.MINIDUMP_TYPE.MiniDumpWithFullMemoryInfo |
+                                       NativeMethods.MINIDUMP_TYPE.MiniDumpWithThreadInfo |
+                                       NativeMethods.MINIDUMP_TYPE.MiniDumpWithTokenInformation;
+                            break;
+                        case DumpTypeOption.Mini:
+                            dumpType = NativeMethods.MINIDUMP_TYPE.MiniDumpWithThreadInfo;
+                            break;
+                    }
 
-                        NativeMethods.MINIDUMP_TYPE dumpType = NativeMethods.MINIDUMP_TYPE.MiniDumpNormal;
-                        switch (type)
+                    // Retry the write dump on ERROR_PARTIAL_COPY
+                    for (int i = 0; i < 5; i++)
+                    {
+                        // Dump the process!
+                        if (NativeMethods.MiniDumpWriteDump(process.Handle, (uint)process.Id, stream.SafeFileHandle, dumpType, ref exceptionInfo, IntPtr.Zero, IntPtr.Zero))
```

Comment:
Does this mean that `dotnet-dump` will still always use the `dbghelp` API? I guess it's easier given we don't know the version of the running app, but this also means we have no testing of createdump on Windows.